### DISCUSSION TITLE
NavBar Menu

### DISCUSSION THREAD

**Discussion by a-Marino** [heroui-inc/heroui#1267] [Feedback]
The NavBar Menu should close when clicking on a Link, its really anoying having to close the menu on mobile when it shoud close by itself

**Comment by isaacdarcilla** [comment]
Do you have solution on this one?

**Reply by a-Marino** [reply]
Nop, i had to make a custom Nav

**Reply by isaacdarcilla** [reply]
My solution was to use Navbar event [(docs)](https://nextui.org/docs/components/navbar#navbar-events). Then just use the onClick somewhere in the `NavbarMenuItem`, `onClick={() => setIsMenuOpen(false)}` to close the navbar (controlled)

```js

 const [isMenuOpen, setIsMenuOpen] = useState(false);

 <Navbar
   isMenuOpen={isMenuOpen}
   onMenuOpenChange={setIsMenuOpen}
 >
    <NavbarContent className="sm:hidden" justify="start">
        <NavbarMenuToggle aria-label={isMenuOpen ? "Close menu" : "Open menu"} />
    </NavbarContent>
   
   ...

 </Navbar>
```

**Reply by AviStudio** [reply]
Yep! This works!
